client/CServerHandler.cpp

```diff
@@ -193,8 +193,16 @@ void CServerHandler::startLocalServerAndConnect(bool connectToLobby)
 		serverRunner.reset(new ServerThreadRunner());
 #endif
 
+	auto si = std::make_shared<StartInfo>();
+
+	auto lastDifficulty = settings["general"]["lastDifficulty"];
+	if (lastDifficulty.isNumber())
+	{
+		si->difficulty = lastDifficulty.Integer();
+	}
```

Comment:
Redundant. Settings are validated, so it is not possible to get anything else in here.
```suggestion
	si->difficulty = lastDifficulty.Integer();
```

Reply:
But we could get null value if nothing was saved before, am I right?

Reply:
No. If this was simple JsonNode (settings have a bit different logic) then this code would've returned default value for integer (0). But not null.

However since this is a setting, it will *always* have a value. Either one that was set by player before or (if not set) - default value as specified in schema.